src/eccommit.h

```diff
@@ -0,0 +1,24 @@
+/**********************************************************************
+ * Copyright (c) 2020 The libsecp256k1-zkp Developers                 *
+ * Distributed under the MIT software license, see the accompanying   *
+ * file COPYING or http://www.opensource.org/licenses/mit-license.php.*
+ **********************************************************************/
+
+#ifndef SECP256K1_ECCOMMIT_H
+#define SECP256K1_ECCOMMIT_H
+
+/** Helper function to add a 32-byte value to a scalar */
+static int secp256k1_ec_seckey_tweak_add_helper(secp256k1_scalar *sec, const unsigned char *tweak);
+/** Helper function to add a 32-byte value, times G, to an EC point */
+static int secp256k1_ec_pubkey_tweak_add_helper(const secp256k1_ecmult_context* ecmult_ctx, secp256k1_ge *p, const unsigned char *tweak);
+
+/** Compute an ec commitment tweak as hash(pubkey, data). */
+static void secp256k1_ec_commit_tweak(unsigned char *tweak32, secp256k1_ge* pubp, secp256k1_sha256* sha, const unsigned char *data, size_t data_size);
```

Comment:
We should document in (all) ec_commit functions that pubp must not be the point at infinity. Or alternatively, correctly deal with that case and declassify only the infinity bit in `ecdsa_sign_inner`. I implemented that here: https://github.com/jonasnick/secp256k1-zkp/commits/ecdsa-s2c-inf Feel free to cherry-pick & squash.